components/st-microelectronics/landpatterns.stanza

```diff
@@ -0,0 +1,183 @@
+;============================= Grid Array Packages ============================
+;==============================================================================
+
+public pcb-landpattern UFBGA64:
+  make-bga-landpattern(8, 8, 0.5, 0.28, tol(5.0, 0.15), tol(5.0, 0.15), FullMatrix())
+
+public pcb-landpattern UFBGA132:
+  make-bga-landpattern(12, 12, 0.5, 0.28, tol(7.0, 0.05), tol(7.0, 0.05), ThermallyEnhancedMatrix(
+    5 through 8, 5 through 8,
+    6 through 7, 6 through 7,
+  ))
+
+public pcb-landpattern LFBGA100:
+  make-bga-landpattern(10, 10, 0.8, 0.5, tol(10.0, 0.15), tol(10.0, 0.15), FullMatrix())
+
+public pcb-landpattern LFBGA144:
+  make-bga-landpattern(12, 12, 0.8, 0.4, tol(10.0, 0.1), tol(10.0, 0.1), FullMatrix())
+
+public pcb-landpattern LFBGA354:
+  defn depop? (r:Int, c:Int):
+    (c == 19 and contains?([5, 15], r)) or
+    (c == 16 and contains?([6, 8, 10, 12, 14], r))
+  make-bga-landpattern(19, 19, 0.8, 0.4, tol(16.0, 0.15), tol(16.0, 0.15), depop?)
+
+public pcb-landpattern TFBGA100:
+  make-bga-landpattern(10, 10, 0.8, 0.4, tol(8.0, 0.05), tol(8.0, 0.05), FullMatrix())
+
+public pcb-landpattern TFBGA216:
+  make-bga-landpattern(14, 14, 0.8, 0.4, tol(13.0, 0.15), tol(13.0, 0.15), PerimeterMatrix(7 through 9, 7 through 9))
+
+public pcb-landpattern TFBGA225:
+  make-bga-landpattern(15, 15, 0.8, 0.4, tol(13.0, 0.15), tol(13.0, 0.15), FullMatrix())
+
+public pcb-landpattern TFBGA64:
+  make-bga-landpattern(8, 8, 0.5, 0.3, tol(0.5, 0.15), tol(0.5, 0.15), FullMatrix())
+
+public pcb-landpattern UFBGA100:
+  defn depop? (r:Int, c:Int):
+    (r == 2 and (c == 5 or c == 6)) or
+    (r == 9 and (c == 5 or c == 6)) or
+    (c == 2 and (r == 5 or r == 6)) or
+    (c == 9 and (r == 5 or r == 6)) or
+    ((r >= 3 and r <= 8) and  (c >= 3 and c <= 8))
+  make-bga-landpattern(12, 12, 0.50, 0.25, tol(7.0, 0.05), tol(7.0, 0.05), depop?)
+
+public pcb-landpattern UFBGA121:
+  make-bga-landpattern(11, 11, 0.5, 0.290, tol(6.0, 0.15), tol(6.0, 0.15), FullMatrix())
+
+public pcb-landpattern UFBGA129:
+  defn depop? (r:Int, c:Int):
+    (contains?([0, 12], r) and contains?([3, 4, 8, 9], c)) or
+    (contains?([0, 12], c) and contains?([3, 4, 8, 9], r)) or
+    (contains?([3, 5, 7, 9], r) and contains?([4, 6, 8], c)) or
+    (contains?([4, 6, 8], r) and contains?([3, 5, 7, 9], c))
+  make-bga-landpattern(13, 13, 0.50, 0.29, tol(7.0, 0.15), tol(7.0, 0.15), depop?)
+
+public pcb-landpattern UFBGA144:
+  make-bga-landpattern(14, 14, 0.8, 0.4, tol(10.0, 0.15), tol(10.0, 0.15), FullMatrix())
+
+public pcb-landpattern UFBGA169:
+  make-bga-landpattern(13, 13, 0.5, 0.28, tol(7.0, 0.05), tol(7.0, 0.05), FullMatrix())
+  
+public pcb-landpattern UFBGA73:
+  defn depop? (r:Int, c:Int):
+    switch(r):
+      1: c == 3 or c == 6
+      3: c == 9
+      4: c == 1
+      6: c == 9
+      7: c == 1
+      9: c == 4 or c == 7
+      else:
+        false
+  make-bga-landpattern(9, 9, 0.5, 0.29, tol(5.0, 0.15), tol(5.0, 0.15) depop?)
```

Comment:
missing comma after `tol(5.0, 0.15)`?

Reply:
Good catch, thanks!

Reply:
How come the CI went through? There is no unit test using UFBGA73?

Reply:
There is, this macro is auto-discovered and evaluated. However in stanza, commas are kind of optional, I don't know if it is the case for all commas but it seems that stanza just want a separator and a space is good:
```
stanza repl
stanza> defn f (x, y) : x + y
stanza> println $ f(1 2)
3
stanza> println([1 2] == [1, 2])
true
```

Reply:
Cool! But I prefer to use the commas for clarity.

Reply:
I agree, even though "," is white space in stanza we should use it to keep things separated.